app/web/features/userQueries/useLiteUsers.ts

```diff
@@ -0,0 +1,46 @@
+import { liteUserKey, liteUsersKey } from "features/queryKeys";
+import { RpcError } from "grpc-web";
+import { GetLiteUsersRes, LiteUser } from "proto/api_pb";
+import { useQuery } from "react-query";
+import { service } from "service";
+
+import { userStaleTime } from "./constants";
+
+function useLiteUsers(ids: (number | undefined)[]) {
+  const nonFalseyIds = ids?.filter((id): id is number => !!id);
+  const query = useQuery<GetLiteUsersRes.AsObject, RpcError>({
+    queryKey: liteUsersKey(nonFalseyIds),
+    queryFn: () => service.user.getLiteUsers(nonFalseyIds),
+    staleTime: userStaleTime,
+    enabled: nonFalseyIds.length > 0, // run only if there are valid liteUserIds
+  });
+
+  const usersById =
+    query.isLoading || query?.data?.responsesList === undefined
+      ? undefined
+      : new Map(
+          query?.data?.responsesList.map((response) => [
+            response?.user?.userId,
+            response.user,
+          ])
+        );
+
+  return {
+    ...query,
+    data: usersById,
+  };
+}
+
+// should is returned when stale if subsequent refetch queries fail
```

Comment:
I have trouble understanding what this comment is trying to explain 🙈  please elaborate?

Reply:
Ahah sorry I copied that from the old file. I fixed the grammar now. Basically by adding `retry: false` failed queries will not retry by default and will use cached data.